tests/cases/fourslash/formatTypeAnnotation1.ts

```diff
@@ -0,0 +1,17 @@
+/// <reference path='fourslash.ts' />
+
+////function foo(x: number, y?: string): number {}
+////interface Foo {
+////    x: number;
+////    y?: number;
+////}
+
+format.setOption("insertSpaceBeforeTypeAnnotation", true);
+format.document();
+verify.currentFileContentIs(
+`function foo(x : number, y ?: string) : number { }
+interface Foo {
+    x : number;
+    y ?: number;
```

Comment:
Wait, really?

Reply:
https://github.com/microsoft/TypeScript/blob/d45f0d0c9fb933d49a01d41eaa465940cfd78be0/src/services/formatting/rules.ts#L318

https://github.com/microsoft/TypeScript/blob/d45f0d0c9fb933d49a01d41eaa465940cfd78be0/src/services/formatting/rules.ts#L492-L498

`y ?: number; <- SyntaxKind.PropertySignature`

Reply:
What do you mean? :confused:

Reply:
@DanielRosenwasser  What do you mean? 😕

Reply:
It's a little weird... I guess I don't consider the `?` to be part of the type annotation, so I'm surprised that a person would have to turn off `NoSpaceBeforeQuestionMark` to avoid this behavior?

Reply:
@DanielRosenwasser Not sure that I correctly understood your concern. `NoSpaceBeforeQuestionMark` doesn't use any option to enable/disable it, this rule applies by default

The issue that `?` token can be used to define optional parameter/property,
```ts
function foo(x ?: number) {}
```
So `NoSpaceBeforeQuestionMark` conflicts with the `SpaceBeforeTypeAnnotation`/`NoSpaceBeforeTypeAnnotation`, therefore need to exclude checking type annotations context in `NoSpaceBeforeQuestionMark`. 

If the `insertSpaceBeforeTypeAnnotation` is not enabled then `NoSpaceBeforeTypeAnnotation` will be applied because this rule uses `isOptionDisabledOrUndefined`, which means that the user can set` false` or delete this rule, and in both cases, the spaces will be removed.

If a user enables the option `insertSpaceBeforeTypeAnnotation`, formatted will add space before `" :"` and `" ?:"` as _ColonToken_ and _ColonToken_ + _QuestionToken_ are valid tokens in type annotations.

Or #25155 is not a bug, and if a user the enables option `insertSpaceBeforeTypeAnnotation`, formatted has to add space only before `:` and ignore `?:`.

What do you think?

Reply:
@DanielRosenwasser What do you think, can we close the #25155 without fix?